### PR TITLE
Luminous testing

### PR DESCRIPTION
**Pull request by leseb**

### CODE REVIEW

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-centos7_cluster

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-xenial_cluster

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-dmcrypt_journal_collocation

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-update_docker_cluster

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-docker_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-update_docker_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-bluestore_docker_dmcrypt_journal_collocation

File: tox.ini [hunk 1]
```diff
@@ -99,6 +103,11 @@ changedir=
   update_cluster: {toxinidir}/tests/functional/centos/7/cluster
   bluestore_journal_collocation: {toxinidir}/tests/functional/centos/7/bs-journal-collocation
   bluestore_cluster: {toxinidir}/tests/functional/centos/7/bs-cluster
+  bluestore_dmcrypt_journal: {toxinidir}/tests/functional/centos/7/bs-dmcrypt-dedicated-journal/
+  bluestore_dmcrypt_journal_collocation: {toxinidir}/tests/functional/centos/7/bs-dmcrypt-journal-collocation/
+  bluestore_docker_cluster: {toxinidir}/tests/functional/centos/7/bs-docker-cluster
+  bluestore_docker_dedicated_journal: {toxinidir}/tests/functional/centos/7/bs-docker-cluster-dedicated-journal
+  bluestore_docker_dmcrypt_journal_collocation: {toxinidir}/tests/functional/centos/7/bs-docker-cluster-dmcrypt-journal-collocation
```
Comment (guits) [inline tox.ini]: `tests/functional/centos/7/bs-docker-cluster-dmcrypt-journal-collocation` doesn't exist in your PR

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-bluestore_docker_dmcrypt_journal_collocation

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-dmcrypt_journal

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-dmcrypt_journal_collocation

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-journal_collocation

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-update_docker_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-docker_cluster

**Comment by leseb** [comment]
jenkins test luminous-ansible2.2-bluestore_docker_cluster

**Comment by guits** [comment]
jenkins test this please

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-purge_cluster

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-update_docker_cluster

**Comment by guits** [comment]
jenkins test this please

**Comment by leseb** [comment]
jenkins test this please

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-docker_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-docker_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-dmcrypt_journal_collocation

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-centos7_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-bluestore_docker_dedicated_journal

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-docker_dmcrypt_journal_collocation

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-docker_dedicated_journal

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-docker_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-bluestore_docker_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-bluestore_docker_dmcrypt_journal_collocation

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-update_docker_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-update_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-update_dmcrypt

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-purge_docker_cluster

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-update_docker_cluster

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-update_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-journal_collocation

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-xenial_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-update_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-update_dmcrypt

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-update_docker_cluster

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-xenial_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-docker_dmcrypt_journal_collocation

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-xenial_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-centos7_cluster

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-docker_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-centos7_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-bluestore_journal_collocation

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-update_docker_cluster

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-dmcrypt_journal_collocation

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-update_docker_cluster

File: roles/ceph-osd/tasks/scenarios/dmcrypt-dedicated-journal.yml [hunk 2]
```diff
@@ -23,6 +23,24 @@
     - item.0.get("rc", 0) != 0
     - item.1.get("rc", 0) != 0
     - not osd_auto_discovery
+    - osd_objectstore == 'filestore'
```
Comment (leseb) [inline roles/ceph-osd/tasks/scenarios/dmcrypt-dedicated-journal.yml]: On Luminous, `ceph-disk prepare` default is Bluestore, so we need to change the default of `osd_objectstore` I suppose.

File: roles/ceph-osd/tasks/scenarios/dmcrypt-journal-collocation.yml [hunk 3]
```diff
@@ -18,9 +18,22 @@
     - item.value.partitions|count == 0
     - item.value.holders|count == 0
     - dmcrypt_journal_collocation
+    - osd_objectstore == 'filestore'
```
Comment (leseb) [inline roles/ceph-osd/tasks/scenarios/dmcrypt-journal-collocation.yml]: ditto

File: roles/ceph-osd/templates/ceph-osd-run.sh.j2 [hunk 4]
```diff
@@ -45,6 +40,7 @@ create_dev_list $1
   -e KV_IP={{kv_endpoint}} \
   -e KV_PORT={{kv_port}} \
   {% endif -%}
+  -v /dev:/dev \
```
Comment (leseb) [inline roles/ceph-osd/templates/ceph-osd-run.sh.j2]: Waow, why?

**Pull request merged**